gems/pending/disk/modules/VhdxDisk.rb

```diff
@@ -373,9 +374,13 @@ def region_table_entry(table_offset, count)
     VHDX_REGION_TABLE_ENTRY.decode(@vhdx_file.read(SIZEOF_VHDX_REGION_TABLE_ENTRY))
   end
 
+  def bat(entry)
+    @bat.empty? && process_bat
+    @bat[entry]
+  end
+
```

Comment:
I think it would be cleaner if the `process_bat` method returned the `bat` array, and then have the `bat` method just memoize and return the array:

```ruby
def bat
  @bat ||= process_bat
end
```
Then, the caller would just access the returned array: `bat[entry]`.

Also, `process_bat` depends on `@total_bat_entries` being set by `post_process_metadata`. Are we sure it'll be set before every possible call to `bat` can be made.